### ISSUE TITLE
`plot()` not finding graphviz on Windows if graphviz is installed system wide.

### ISSUE THREAD

**Issue by t-kalinowski** [rstudio/keras#1337]
First reported in Manning book discussion forum by George Fox.

**Comment by ggeeoorrgg** [comment]
Is this problem resolved?
I am triyng to reproduce some examples from the book "Deep learning with R". Calling plot(model) after Listing 7.11 falls with error: "See ?keras::plot.keras.engine.training.Model for  instructions on how to install graphviz and pydot Error in plot(model) :  ImportError: You must install pydot (`pip install pydot`) and install graphviz (see instructions at https://graphviz.gitlab.io/download/) for plot_model to work."
But I have installed pydot, pydotplus, graphviz.
Moreover the same code in Python from its sibling "Deep Learning with Python" (with plot_model(model)) works fine.
Any thoughts?

**Comment by t-kalinowski** [comment]
I can only reproduce this in the RStudio IDE R console, I can't reproduce this in a terminal (in or out of the IDE). I don't think this is a reticulate issue (at least, not directly). 

To reproduce locally:
```
library(reticulate)
# virtualenv_create("rstudio-find-graphviz", python = install_python(), packages = "pydot")
use_virtualenv("rstudio-find-graphviz")
py_run_string("import pydot; d = pydot.Dot(); d.create()")
```
Gives output in the IDE:
```
  OSError: [WinError 6] The handle is invalid
```

but both these succeed in terminal:
```
C:/Users/kalin/Documents/.virtualenvs/rstudio-find-graphviz/Scripts/python.exe -c "import pydot; d = pydot.Dot(); print(d.create())"

C:/PROGRA~1/R/R-42~1.2/bin/Rscript.exe -e "library(reticulate); use_virtualenv('rstudio-find-graphviz'); py_run_string('import pydot; d = pydot.Dot(); print(d.create())')
```

**Comment by t-kalinowski** [comment]
@kevinushey do you have any guesses about why the code in the IDE might return that error?

**Comment by t-kalinowski** [comment]
Making sure the graphviz is on the PATH in the IDE doesn't fix the issue. 

i.e., downloading from https://www.graphviz.org/download/ and then running:
```r

PATH <- Sys.getenv("PATH")
PATH <- paste0(c(
  r"(C:\Users\kalin\Documents\windows_10_msbuild_Release_graphviz-7.1.0-win32\Graphviz\bin)",
  PATH),
  collapse = .Platform$path.sep)
Sys.setenv(PATH = PATH)

library(reticulate)
py_run_string("import pydot; d = pydot.Dot(); d.create()")
```

**Comment by kevinushey** [comment]
This might be related:

https://github.com/rstudio/reticulate/issues/518

**Comment by ggeeoorrgg** [comment]
> I can only reproduce this in the RStudio IDE R console...
> ```
Yes, you are most likely right. In my case the issue occurs in RStudio IDE. I ran this code in another IDE (VSC) and it works OK. So this is certainly a bug in RStudio.

**Comment by t-kalinowski** [comment]
Fixed on main now.

**Comment by ggeeoorrgg** [comment]
For some reason **utils.py** hasn`t included in **keras 2.11.1** Windows binaries on CRAN. So after library updating this issue... occurs again.

**Comment by t-kalinowski** [comment]
Hi @ggeeoorrgg, this fix for this isn't on CRAN yet. It should go out in the next 2 weeks or so. For now you can install the development version of keras:
```R
remotes::install_github("rstudio/keras")
```
